### ISSUE TITLE
Nullability not tracked for is/cast pattern in switch

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#26737]
**Version Used**: Nullable Reference Types Preview (7 May 2018)

**Steps to Reproduce**:

```csharp
object? obj = null;
if (obj is Type)
{
  ((Type)obj).ToString();
}
```

**Expected Behavior**:

No nullability-related warning is reported for the expression `((Type)obj)`.

**Actual Behavior**:

CS8600 and CS8602 are reported for `((Type)obj)`.

**Comment by jcouv** [comment]
Conversely, `s is null` should also inform nullability.

**Comment by jcouv** [comment]
Patterns in `is` operator are handled by https://github.com/dotnet/roslyn/pull/28686
Patterns in `switch` will be handled separately.

**Comment by jcouv** [comment]
The remaining issue (nullability in `switch`) is tracked in more than one issue. Closing this in favor of https://github.com/dotnet/roslyn/issues/23944
